numba/tests/test_slices.py

```diff
@@ -78,6 +82,62 @@ def test_slice_constructor(self):
             got = cfunc(*args)
             self.assertPreciseEqual(got, expected)
 
+    def test_slice_indices(self):
+        """Test that a numba slice returns same result for .indices as a python one."""
+        slices = starmap(
+            slice,
+            product(
+                chain(range(-5, 5), (None,)),
+                chain(range(-5, 5), (None,)),
+                chain(range(-5, 5), (None,))
+            )
+        )
+        lens = range(-2, 3)
+
+        cfunc = jit(nopython=True)(slice_indices)
+
+        for s, l in product(slices, lens):
```

Comment:
This loop is a bit tricky to read and understand a mixes several concerns. It would be good to simplify this. But I am not yet sure how to make this more legible.  How about the following:

```
            if l < 0 and not utils.IS_PY3:
                # Passing a negative length to slice.indices in python2 is
                # undefined. See https://bugs.python.org/issue14794#msg174678
                continue
            try:
                expected = slice_indices(s, l)
            except Exception as e:
                with self.assertRaises(type(e)):
                    cfunc(s, l)
            else:
                    self.assertPreciseEqual(expected, cfunc(s, l))
```
That way it is a bit easier to read the flow:
* First, skip the length less than zero on Python 2 entirely
* Then try to compare Numba and CPython, if it's an exception, make sure the same on is raised and otherwise check the result.

Makes sense?

Reply:
Yup, your way is more legible. I've switched to this.